### ISSUE TITLE
Floating IP support for DigitalOcean

### ISSUE THREAD

**Issue by andrewvaughan** [ansible/ansible#20137]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
digitalocean

##### SUMMARY
Add Floating IP Support to DigitalOcean Module

##### DETAILS
Parroting [a previous request](https://github.com/ansible/ansible-modules-core/issues/3004) (in the old issue platform) to add support for DigitalOcean's Floating IP service to allow for Floating IPs to be generated and/or assigned to servers.  This will empower Ansible to support High Availability setups that use DigitalOcean's services.

```yaml
- name: Generate a new Floating IP in the nyc1 datacenter
  digital_ocean: >
    command=floating_ip
    state=present
    region_id=nyc1
    api_token={{ do_token }}
  register: new_floating_ip

- name: Assign an existing Floating IP to an existing droplet
  digital_ocean: >
    command=floating_ip
    state=present
    name={{ do_droplet.droplet.name }}
    floating_ip={{ new_floating_ip.floating_ip.ip }}
    api_token={{ do_token }}
  register: assigned_floating_ip

- name: Assign a new Floating IP to an existing droplet
  digital_ocean: >
    command=floating_ip
    state=present
    name={{ do_droplet.droplet.name }}
    api_token={{ do_token }}
  register: new_floating_ip

- name: Assign an existing Floating IP to a new droplet
  digital_ocean: >
    command=droplet
    state=present
    name=my-new-server
    size_id=1gb
    region_id=nyc1
    image_id=ubuntu-16-10-x64
    floating_ip={{ new_floating_ip.floating_ip.ip }}
    api_token={{ do_token }}
  register: existing_server

- name: Assign a new Floating IP to a new droplet
  digital_ocean: >
    command=droplet
    state=present
    name=my-second-server
    size_id=1gb
    region_id=nyc1
    image_id=ubuntu-16-10-x64
    floating_ip=new
    api_token={{ do_token }}
  register: new_server

- name: Remove an existing Floating IP
  digital_ocean: >
    command=floating_ip
    state=absent
    floating_ip={{ assigned_floating_ip.floating_ip.ip }}
    api_token={{ do_token }}
```

There may be more/less functionality depending on standards - the scheme above is a best-effort at what this functionality might look like.

**Comment by BondAnthony** [comment]
PR #18922 addresses this

**Comment by abadger** [comment]
#18922 has been merged for the ansible 2.4.0 release.  Closing this ticket.
